File: app/templates/app-java/src/main/java/ui/main/MainFragment.java
package <%= appPackage %>.ui.main;

import <%= appPackage %>.di.components.MainComponent;
import <%= appPackage %>.ui.base.BaseFragment;
import <%= appPackage %>.ui.base.EmptyPresenter;
import <%= appPackage %>.R;

public class MainFragment extends BaseFragment<EmptyPresenter> {

    @Override
    protected void inject() {
        getComponent(MainComponent.class).inject(this);
    }

    @Override
    protected int getLayoutResource() {
        return R.layout.fragment_main;
    }

    @Override
    protected EmptyPresenter getPresenter() {
        return new EmptyPresenter();
    }

}

File: app/templates/app-kotlin/src/main/java/ui/main/MainFragment.kt
package <%= appPackage %>.ui.main

import <%= appPackage %>.di.components.MainComponent
import <%= appPackage %>.ui.base.BaseFragment
import <%= appPackage %>.ui.base.EmptyPresenter
import <%= appPackage %>.R


class MainFragment : BaseFragment<EmptyPresenter>() {

    override fun inject() {
        getComponent(MainComponent::class.java).inject(this)
    }

    override fun getLayoutResource(): Int {
        return R.layout.fragment_main
    }

    <% if (nucleus == false) { %>override fun getPresenter(): EmptyPresenter {
        return EmptyPresenter()
    }<% } %>
}
